### DISCUSSION TITLE
How to programmatically add stylable attribute to a view?

### DISCUSSION THREAD

**Discussion by Kage0923** [community/community#42475] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

How to programmatically add stylable attribute to a view?
If anyone answer about this question, I will hire him for big project.

**Comment by TechSavvy6** [comment, marked as answer]
<declare-styleable name="MBButton">
    <attr name="button_type_accessible" format="enum">
        <enum name="primary" value="0" />
        <enum name="secondary" value="1" />
        <enum name="secondary_alternative" value="2" />
        <enum name="ghost" value="3" />
        <enum name="ghost_alternative" value="4" />
        <enum name="action" value="5" />
    </attr>
    <attr name="is_medium_size_accessible" format="boolean" />
    <attr name="is_small_size_accessible" format="boolean" />
    <attr name="make_accessible_button" format="string" />
</declare-styleable>

This is the constructor of my custom button

```
class CustomButton @JvmOverloads constructor(
context: Context,
attrs: AttributeSet? = null,
defStyleAttr: Int = R.attr.mbButtonSecondaryStyle
)
```
